### DISCUSSION TITLE
Can't create right certificate for Adobe Animate AIR

### DISCUSSION THREAD

**Discussion by ZhorovEgor** [airsdk/Adobe-Runtime-Support#3112] [Q&A]
Howdy guys! When publishing on the iOS version of Adobe Animate, I get the following error:
![Screenshot_2](https://github.com/airsdk/Adobe-Runtime-Support/assets/63095086/ccc5eb0f-3a86-48c7-ba79-1e67d3b298f9)

As I understand it, the program has claims against my p12 certificate, right? If so, I registered it through an Apple developer account. Did I do it right? I was guided by this video: https://www.youtube.com/watch?v=AZzi71xs7_s

If another certificate is required, could you provide a link to a resource on how to issue it for Animate?

**Comment by tonystech** [comment]
At 1 minute and 50 seconds, they right click on the certificate and export it as a p12 file. Before doing that, click on the dropdown arrow to the left of the certificate icon. This will reveal your private key. Hold shift and select both the certificate and the private key. Now right click and export it as a p12 file. 

This was something we had to do years ago so it might not be a thing anymore, but back in the day, selecting just the certificate and exporting it didn't include the private key. I do it out of habit so I'm not sure if it's necessary anymore or not. Give it a try. 

I hope that helps!

**Comment by ajwfrost** [comment]
I just looked at that video .. it's showing you how to create a certificate for Apple Push Services, so I would expect this probably isn't issued via the same certificate chain as we're expecting for an application.

The steps to create the certificate signing request look fine, but in the Developer portal, go to "Certificates, Identifiers & Profiles" and then go to "Certificates" and hit the blue "+" button to go to "Create a New Certificate".

Then you would need to create a certificate for development - where you use it for your specific devices with debugging etc - and a certificate for distribution if you're looking to use this for the App Store etc. For me, it's currently showing two 'distribution' options, one is just "Apple Distribution" and one is "iOS Distribution".. I would suggest the "Apple" ones are better, but am not 100% sure on the differences... they change all this and the various options here, surprisingly regularly (as you may find if you look on Google for "how to" guides, most of them list options that aren't there now...)

Hope that helps!

thanks
